src/celeritas/optical/detail/OpticalGenAlgorithms.cc

```diff
@@ -8,16 +8,31 @@
 #include "OpticalGenAlgorithms.hh"
 
 #include <algorithm>
+#include <numeric>
 
 #include "corecel/Assert.hh"
 
 namespace celeritas
 {
 namespace detail
 {
+//---------------------------------------------------------------------------//
+struct AccumNumPhotons
+{
+    // Accumulate the number of optical photons from the distribution data
+    CELER_FUNCTION size_type
+    operator()(size_type count,
+               celeritas::optical::GeneratorDistributionData const& data) const
+    {
+        return count + data.num_photons;
+    }
+};
+
 //---------------------------------------------------------------------------//
 /*!
  * Remove all invalid distributions from the buffer.
+ *
+ * This returns the total number of valid distributions in the buffer.
```

Comment:
could use the doxygen markup here 😉 
```suggestion
 *
 * \return the total number of valid distributions in the buffer
```